contentcuration/contentcuration/frontend/channelEdit/vuex/assessmentItem/mutations.js

```diff
@@ -2,25 +2,49 @@ import Vue from 'vue';
 import { mergeMapItem } from 'shared/vuex/utils';
 
 export function ADD_ASSESSMENTITEM(state, assessmentItem) {
+  UPDATE_ASSESSMENTITEM(state, assessmentItem);
+}
+
+export function UPDATE_ASSESSMENTITEM(state, assessmentItem) {
   if (!assessmentItem.assessment_id) {
     throw ReferenceError('assessment_id must be defined to update an assessment item');
   }
   if (!assessmentItem.contentnode) {
     throw ReferenceError('contentnode must be defined to update an assessment item');
   }
-  state.assessmentItemsMap[assessmentItem.contentnode] = mergeMapItem(
-    state.assessmentItemsMap[assessmentItem.contentnode] || {},
-    assessmentItem,
-    'assessment_id'
-  );
-}
 
-export function ADD_ASSESSMENTITEMS(state, assessmentItems = []) {
-  assessmentItems.forEach(assessmentItem => {
-    ADD_ASSESSMENTITEM(state, assessmentItem);
-  });
+  // data can come from API that returns answers and hints as string
+  let answers, hints;
+  if (typeof assessmentItem.answers === 'string') {
```

Comment:
Yeah, lets update the API to return this properly (but we can do as a follow up).

Reply:
https://github.com/learningequality/studio/issues/1855